javaparser-core/src/main/java/com/github/javaparser/ast/validator/language_level_validations/Java14PreviewValidator.java

```diff
@@ -39,7 +39,7 @@ public Java14PreviewValidator() {
         {
             // first preview within Java 14 - https://openjdk.java.net/jeps/359
             remove(noRecordDeclaration);
-            add(recordAsTypeIdentifierNotAllowed);
+            add(recordAsClassAndInterfaceIdentifierNotAllowed);
```

Comment:
Please revert this back to `TypeIdentifier` - `record` is not permitted e.g. as an enum`s identifier either.